tests/test_suites/ShortSystemTests/test_metadata_load_legacy.sh

```diff
@@ -0,0 +1,46 @@
+timeout_set 30 seconds
+master_cfg+="MAGIC_DEBUG_LOG = $TEMP_DIR/syslog|LOG_FLUSH_ON=DEBUG"
+touch "$TEMP_DIR/syslog"
+
+CHUNKSERVERS=0 \
+	MASTERSERVERS=1 \
+	USE_RAMDISK="YES" \
+	MASTER_EXTRA_CONFIG="${master_cfg}" \
+	MOUNT_EXTRA_CONFIG="sfscachemode=NEVER" \
+	setup_local_empty_saunafs info
+
+metadata_file="${info[master_data_path]}/metadata.sfs"
+
+# Access mountpoint to create metadata
+cd "${info[mount0]}"
+for i in {1..100}; do
+	touch file$i
+	mkdir dir$i
+done
+
+# Exit mount point and save metadata
+cd "${TEMP_DIR}"
+assert_success saunafs_master_daemon stop
+
+# Expected Values
+assert_equals "SFSM 2.9" "$(head -c8 ${metadata_file})"
+assert_equals "[SFS EOF MARKER]" "$(tail -c16 ${metadata_file})"
+
+# Modify metadata signature for legacy and reload metadata by staring master
+echo -n "LIZ" | dd of=${metadata_file} bs=1 seek=0 count=3 conv=notrunc
+echo -n "MFS" | dd of=${metadata_file} bs=1 seek=$(( $(stat --print="%s" ${metadata_file}) - 15)) count=3 conv=notrunc
+assert_equals "LIZM 2.9" "$(head -c8 ${metadata_file})"
+assert_equals "[MFS EOF MARKER]" "$(tail -c16 ${metadata_file})"
+
+mv -iv "${info[master_data_path]}/metadata.sfs" "${info[master_data_path]}/metadata.mfs"
+
+# Start master server to reload metadata legacy format
+assertlocal_success saunafs_master_daemon start
+# Restart master server to load metadata in new format when both are present
```

Comment:
Does It means that metadata will be always converted to new format and saved to metadata.sfs?
Why there are two metadatas here with different formats?

Reply:
Yes, so the logic is like this. If only `metadata.mfs` is found, it loads it, and on the next metadata dump the new `metadata.sfs` is created. This means at this point both files will be present until system admin manually deletes the legacy `metadata.mfs`. The system detect if both files are present and loads the actual `metadata.sfs` in this case. Also when SaunaFS shadows connect to legacy master cluster, it downloads the metadata content to the file called `metadata.sfs` but with wrong tags internally until next dump occurs, then internal tags are properly written to file. This how ever was not yet tested in this integration tests, because it requires the testing framework to support mixing actual and legacy binaries and goes beyond the scope of this PR.

Reply:
Clear now.